edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/edgeagent/EdgeAgent.kt

```diff
@@ -919,70 +921,7 @@ open class EdgeAgent {
-            }
-            if (!requestPresentationJson.containsKey("from")) {
-                throw EdgeAgentError.MissingOrNullFieldError("from", "Request")
-            }
-
-            val requestId = requestPresentationJson["id"]!!
-            val requestBody = requestPresentationJson["body"]!!
-            val requestAttachments = requestPresentationJson["attachments"]!!
-            val requestThid = requestPresentationJson["thid"]!!
-            val requestFrom = requestPresentationJson["from"]!!
-
-            if (requestAttachments.jsonArray.size == 0) {
-                throw EdgeAgentError.MissingOrNullFieldError("attachments", "Request")
-            }
-            val attachmentJsonObject = requestAttachments.jsonArray[0]
-            if (!attachmentJsonObject.jsonObject.containsKey("id")) {
-                throw EdgeAgentError.MissingOrNullFieldError("id", "Request attachments")
-            }
-            if (!attachmentJsonObject.jsonObject.containsKey("media_type")) {
-                throw EdgeAgentError.MissingOrNullFieldError("media_type", "Request attachments")
-            }
-            if (!attachmentJsonObject.jsonObject.containsKey("data")) {
-                if (!attachmentJsonObject.jsonObject["data"]!!.jsonObject.containsKey("json")) {
-                    throw EdgeAgentError.MissingOrNullFieldError("json", "Request attachments data")
-                }
-                throw EdgeAgentError.MissingOrNullFieldError("data", "Request attachments")
-            }
-            if (!attachmentJsonObject.jsonObject.containsKey("format")) {
-                throw EdgeAgentError.MissingOrNullFieldError("format", "Request attachments")
-            }
-            val attachmentId = attachmentJsonObject.jsonObject["id"]!!
-            val attachmentMediaType = attachmentJsonObject.jsonObject["media_type"]!!
-            val attachmentData = attachmentJsonObject.jsonObject["data"]!!.jsonObject["json"]!!
-            val attachmentFormat = attachmentJsonObject.jsonObject["format"]!!
-
-            val attachmentDescriptor = AttachmentDescriptor(
-                id = attachmentId.jsonPrimitive.content,
-                mediaType = attachmentMediaType.jsonPrimitive.content,
-                data = AttachmentJsonData(attachmentData.toString()),
-                format = attachmentFormat.jsonPrimitive.content
-            )
-
-            val requestPresentation = RequestPresentation(
-                id = requestId.jsonPrimitive.content,
-                body = Json.decodeFromString(requestBody.jsonObject.toString()),
-                attachments = arrayOf(attachmentDescriptor),
-                thid = requestThid.jsonPrimitive.content,
-                from = DID(requestFrom.jsonPrimitive.content),
-                to = ownDID
-            )
-
-            pluto.storeMessage(requestPresentation.makeMessage())
+            connectionlessInvitation(ownDID, jsonString)
```

Comment:
As we were talking yesterday this is not needed

Reply:
You need to parse and return the Credential Offer message

Reply:
I have changed the implementation. Please take another look.